OpenRA.Game/FileFormats/Png.cs

```diff
@@ -94,8 +94,8 @@ public Png(Stream s)
 
 						case "PLTE":
 						{
-							Palette = new Color[256];
-							for (var i = 0; i < length / 3; i++)
+							Palette = new Color[length / 3];
```

Comment:
Already prior to this change litle to no validation is done on the header fields. This is somewhat a security concern. Bad maps could cause the game to crash.

Reply:
We should file a separate issue if we want to handle malformed PNGs more gracefully.